pkg/hnsw/visited.go

```diff
@@ -0,0 +1,42 @@
+package hnsw
+
+import "fmt"
+
+type Set struct {
```

Comment:
What's the benefit of this type over `map[NodeID]interface{}` directly?

```
x := map[NodeID]interface{}
len(x)
```

works to fetch the size.

Reply:
Yeah, I did this to maintain readability when I implemented the algorithm. I can remove this abstraction 👍